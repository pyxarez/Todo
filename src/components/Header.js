import React from 'react';

import './Header.res/style.css';

Header.propTypes = {
  title: React.PropTypes.string.isRequired,
  children: React.PropTypes.oneOfType(
    [
      React.PropTypes.arrayOf(React.PropTypes.element),
      React.PropTypes.bool,
      React.PropTypes.object
    ]
  ),
}

export default function Header({title, children}) {
  return (
    <header className="my-header-component">
      <h1 className="title">{title}</h1>
    {children}
    </header>
  );
}
