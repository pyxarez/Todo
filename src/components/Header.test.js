import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

const setup = () => {
  const props = {
    title: 'Todo'
  };

  const wrapper = shallow(
    <Header {...props}>
      <div className="testClass" />
    </Header>
  );

  return {
    props,
    wrapper
  }
};

describe('Components :: Header', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});