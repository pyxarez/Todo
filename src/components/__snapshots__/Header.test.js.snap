exports[`Components :: Header should render self and subcomponents 1`] = `
<header
  className="my-header-component">
  <h1
    className="title">
    Todo
  </h1>
  <div
    className="testClass" />
</header>
`;
